app/models/fe/element.rb

```diff
@@ -110,10 +110,10 @@ def previous_element(question_sheet, page = nil)
     end
 
     def hidden_by_conditional?(answer_sheet, page, prev_el)
-      (prev_el ||= previous_element(answer_sheet.question_sheet, page)) &&
-        prev_el.is_a?(Fe::Question) &&
-        prev_el.conditional == self &&
-        !prev_el.conditional_match(answer_sheet)
+      !!((prev_el ||= previous_element(answer_sheet.question_sheet, page)) &&
```

Comment:
Doesn't it just change null and false to false, and make everything else true?  I was going for a true boolean return value.  I'm happy to use more intermediate steps though, just wondering if my understanding of double bang is on.. it doesn't seem to confusing other than looking weird at first to me.